wolfssl/ssl.h

```diff
@@ -529,7 +529,9 @@ struct WOLFSSL_BIO {
     int          wrIdx;         /* current index for write buffer */
     int          rdIdx;         /* current read index */
     int          readRq;        /* read request */
-    int          num;           /* socket num or length */
+    int          len;           /* length of buffered data */
```

Comment:
I am fairly certain this will break some openssl compatibility since `num` is an expected dereference-able member. Can you keep the name `num` and still separate out the `fd`?

Reply:
Like most OpenSSL structs, `BIO` is an opaque pointer so there is no way to access its the members. Its defined like this in my system headers `typedef struct bio_st BIO;`. There is no need to worry about breaking compatibility.

Reply:
Fairly certain older openssl's allowed direct BIO deference... even though it is a terrible terrible thing to do. Is there a reason we cannot leave the `num` as the member name?

Reply:
In OpenSSL 1.0.2h, the bio.h that gets installed has this:

```
typedef struct bio_st BIO;
.
.
.
struct bio_st {
    BIO_METHOD *method;
    /* bio, mode, argp, argi, argl, ret */
    long (*callback) (struct bio_st *, int, const char *, int, long, long);
    char *cb_arg;               /* first argument for the callback */
    int init;
    int shutdown;
    int flags;                  /* extra storage */
    int retry_reason;
    int num;
    void *ptr;
    struct bio_st *next_bio;    /* used by filter BIOs */
    struct bio_st *prev_bio;    /* used by filter BIOs */
    int references;
    unsigned long num_read;
    unsigned long num_write;
    CRYPTO_EX_DATA ex_data;
};
```

So the struct isn't opaque in all OpenSSL versions.

Reply:
> `bio->num` is removed to catch any configurations where `bio->num` hasn't been updated to the appropriate value

I wanted to change the name to make sure that it isn't referenced in any exotic configuration. Also I feel like `len` better describes how the member is used.

Reply:
This continues all the way to openssl 1.0.2u

Reply:
Hmm so refactoring this into two members shouldn't happen in the first place? Do we want to maintain compatibility with such an old version of OpenSSL?

Reply:
@julek-wolfssl  Yes we do need to support older openssl uses. I believe the older openssl also uses this for file descriptor as we are. I like this cleanup, but not sure if we can do it without breaking any backwards compatibility. If you keep this PR, then please change back to `num` and keep the FD separation. Thanks, David G.

Reply:
Closing as the refactor is not possible. Thanks for the input everyone.